pennylane/ftqc/lattice.py

```diff
@@ -0,0 +1,169 @@
+        return self._graph
+
+
+class LatticeShape(Enum):
+    """Enum to define valid set of lattice shape supported."""
+
+    chain = auto()
+    square = auto()
+    rectangle = auto()
+    triangle = auto()
+    honeycomb = auto()
+    cubic = auto()
+
+
+# map between lattice name and dimensions
+_LATTICE_DIM_MAP = {
+    "chain": 1,
+    "square": 2,
+    "rectangle": 2,
+    "cubic": 3,
+    "triangle": 2,
+    "honeycomb": 2,
+}
+
+# map between lattice name and networkx method name
+_LATTICE_GENERATOR_MAP = {
+    "chain": "grid_graph",
+    "square": "grid_graph",
+    "rectangle": "grid_graph",
+    "cubic": "grid_graph",
+    "triangle": "triangular_lattice_graph",
+    "honeycomb": "hexagonal_lattice_graph",
+}
+
+
+@lru_cache
+def _supported_shapes():
+    r"""Return the supported shape in str"""
+    return [shape.name for shape in LatticeShape]
+
+
+def generate_lattice(dims: Sequence[int], lattice: str) -> Lattice:
+    r"""Generates a :class:`~pennylane.ftqc.Lattice` object with a given geometric parameters and its shape name.
+
+    Args:
+        dims(List[int]): Geometric parameters for lattice generation. For lattices generated by `nx.grid_graph` ( ``'chain'``, ``'rectangle'``,  ``'square'``, ``'cubic'``),
+        `dims` contains the number of nodes in the each direction of grid. Per ``'honeycomb'`` or ``'triangle'``, the generated lattices will have dims[0] rows and dims[1]
+        columns of hexagons or triangles.
+        lattice (str): Shape of the lattice. Input values can be ``'chain'``, ``'square'``, ``'rectangle'``, ``'honeycomb'``, ``'triangle'``, ``'cubic'``.
+
+    Returns:
+        a :class:`~pennylane.ftqc.Lattice` object.
+
+    Raises:
+        ValueError: If the lattice shape is not supported or the dimensions are invalid.
+    """
+
+    lattice_shape = lattice.strip().lower()
+    supported_shapes = _supported_shapes()
+
+    if lattice_shape not in supported_shapes:
+        raise ValueError(
+            f"Lattice shape, '{lattice}' is not supported."
+            f"Please set lattice to: {supported_shapes}."
+        )
+
+    if _LATTICE_DIM_MAP[lattice_shape] != len(dims):
+        raise ValueError(
+            f"For a {lattice_shape} lattice, the length of dims should be {_LATTICE_DIM_MAP[lattice_shape]} instead of {len(dims)}"
+        )
+
+    lattice_generate_method = getattr(nx, _LATTICE_GENERATOR_MAP[lattice_shape])
+
+    if _LATTICE_GENERATOR_MAP[lattice_shape] == "grid_graph":
+        lattice_obj = Lattice(lattice_shape, lattice_generate_method(dims))
+        return lattice_obj
+
+    if _LATTICE_GENERATOR_MAP[lattice_shape] in [
+        "triangular_lattice_graph",
+        "hexagonal_lattice_graph",
+    ]:
+        lattice_obj = Lattice(lattice_shape, lattice_generate_method(dims[0], dims[1]))
+        return lattice_obj
+
+    raise ValueError(
+        f"The given {lattice_shape} lattice shape and geometric parameters {dims} are not supported."
+    )  # pragma : no cover
```

Comment:
I *think* this has to be on the same line as `raise ValueError` to make codecov happy